docs/testing.md

```diff
@@ -1 +1,109 @@
+
+Your test suites will be composed of test "specs" - individual tests that should examine one functional piece of the module you're testing. We have it designed so that a single suite should correspond with a single module. These can all be found under test/specs in the root of this repo, and should roughly follow the same directory structure as the original source code location for clarity and consistency.
+
+Each suite works in a couple components.  
+**1. Load the module(s) you're testing using RequireJS.**
+
+We use RequireJS to load modules. The set of paths and namespaces is available in functional-site/js/require-config.js (and possibly could be published elsewhere). As an example, to load the kbaseConfig module for testing, this syntax is used to wrap a test suite:
+
+```
+'use strict';
+define(['kb.config'], function(Config) {
+    ... specs go here ...
+});
+```
+
+**2. Use ```describe``` to start the suite.**
+
+Next, we use Jasmine's ```describe``` function to start the suite.
+
+```
+'use strict';
+define(['kb.config'], function(Config) {
+    describe('The KBase Config object', function() {
+        ... specs go here ...
+    });
+});
+```
+
+You can have multiple ```describe``` statements in a single file. Just note that they do not interact with each other, so any variables you declare in one will not be available in the other.
+
+**3. Use ```it``` to define a test.**
+
+Finally, individual test specs are just calls to the Jasmine's ```it``` function. Like ```describe```, these take a descriptive string, and a function that runs the test.
+
+```
+'use strict';
+define(['kb.config'], function(Config) {
+    describe('The KBase Config object', function() {
+        it('should have a Workspace url', function() {
+            var wsUrl = Config.hasItem('services.workspace.url');
+            expect(wsUrl).toBe(true);
+        });
+    });
+});
+```
+
+There are a few things going on inside of the spec. The main testing portion is that ```expect``` function, which takes a value called the "actual." This gets chained with a Matcher function that takes the expected value. In the example above, we ```expect``` the wsUrl (actually returned from the function) simply to be defined (in the JavaScript definition of the term).
+
+There are a large number of Matcher functions available for most cases, and they can be prepended with a ```.not.``` (e.g. ```expect(false).not.toBe(true)```) You can see these (with examples!) in the Jasmine Documentation [here](http://jasmine.github.io/2.3/introduction.html#section-Included_Matchers)
+
+Any variables or constants you might need across tests can be defined outside. The scoping rules mean that each spec has access to variables outside of them.
```

Comment:
Any variables or constants you might need across tests can be defined outside the tests (i.e., before any 'describe's). The scoping rules mean that each spec has access to the global variables that are defined in the same file but outside the tests.